### PR TITLE
Set class and tests

### PR DESCRIPTION
**Pull request by sdweber422**

### CODE REVIEW

File: spec/set.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import chai, { expect, assert } from 'chai'
+import chaiChange from 'chai-change'
+import sinon from 'sinon'
+import Set from '../src/set'
+
+chai.use(chaiChange)
+
+describe.only( 'Set', () => {
```
Comment (shakalee14) [inline spec/set.js]: can we remove the .only method from this describe statement before merging to master?

File: spec/set.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+  let mySet, myOtherSet, logger, logg
```
Comment (shakalee14) [inline spec/set.js]: this may be more appropriate to declare these as ```const```

File: spec/set.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    logg.restore()
+  })
+
+  it( 'exists', () => {
+    expect( Set ).to.be.a( 'function' )
+  })
+
+  context( '#add( element )', () => {
+
+    it( 'should add element to the Set', () => {
```
Comment (shakalee14) [inline spec/set.js]: small typo ``` it( ' should add an element to the Set' , () => { ... } ) ```

File: src/set.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+  isEmpty() {
+    return this.size() === 0
+  }
+
+  contains( element ) {
+    return this._data.includes( element )
+  }
+
+  remove( element ) {
```
Comment (shakalee14) [inline src/set.js:24]: nice!

File: src/set.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      throw new Error( 'Element does not exist in set' )
+    }
+    this._data = this._data.filter( data => data !== element )
+  }
+
+  forEach( givenFunction ) {
+    if ( !givenFunction ) {
+      throw new Error( 'No function provided' )
+    }
+    try {
```
Comment (shakalee14) [inline src/set.js]: nice use of try and catch :)

File: src/set.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      if ( this._data.includes( element ) ) {
+        memo.push( element )
+      }
+      return memo
+    }, [] ).sort( ( element, nextElement )  => {
+      return element > nextElement
+    })
+  }
+
+  difference( otherSet ) {
```
Comment (shakalee14) [inline src/set.js:72]: are you going to complete this function?

File: src/set.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+  clone() {
+    return new Set( this._data )
+  }
+
+}
+
+module.exports = Set
+const mySet = new Set( [ 5, 6, 7, 8 ] )
+console.log('mySet._data', mySet._data)
```
Comment (shakalee14) [inline src/set.js]: do we need all these console.logs at the end of the file? i suspect CDD (console log driven development)

File: spec/set.js [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+import chaiChange from 'chai-change'
+import Set from '../src/set'
+
+chai.use(chaiChange)
+
+describe( 'Set', () => {
+
+  let mySet, myOtherSet, forEachTestArray
+
+  beforeEach( () => {
```
Comment (Anasauce) [inline spec/set.js:11]: nice! Good implementation of this is testing 💪

File: src/set.js [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    }, [] ).sort( ( element, nextElement )  => {
+      return element > nextElement
+    })
+  }
+
+  difference( otherSet ) {
+    if( !otherSet || !Array.isArray( otherSet ) ) {
+      return this._data
+    }
+    return this._data.reduce( ( memo, element ) => {
```
Comment (Anasauce) [inline src/set.js:76]: nice use of reduce!

File: src/set.js [hunk 10]
```diff
@@ -0,0 +1,95 @@
+'use strict'
+
+export default class Set {
+
+  constructor( data = [] ) {
+    this._data = data
```
Comment (Anasauce) [inline src/set.js:6]: why the use of ```this._data```? How does this help you with you Set implementation?

**Review by Anasauce — COMMENTED** [review]
Good, clean, concise code! Good job!

**Pull request merged**